app/extensions/email/__init__.py

```diff
@@ -105,63 +157,173 @@ def template(self, template, **kwargs):
-                minified_css_html,
-                remove_comments=True,
-                remove_empty_space=True,
-                remove_all_empty_space=True,
-            )
-
-            # Add web fonts
-            webfonts = [
-                '<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Assistant:200|Chango|Molle:400i&display=swap">',
-                '<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">',
-                '<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Code+Pro&display=swap">',
-            ]
-            webfonts_html = ''.join(webfonts)
-            minified_html = minified_html.replace(
-                '</head>', '%s</head>' % (WEBFONTS_PLACEHOLDER_CODE,)
-            )
-            final_html = minified_html.replace(WEBFONTS_PLACEHOLDER_CODE, webfonts_html)
-
-            with open('email.latest.html', 'w') as temp:
-                temp.write(final_html)
-
-            self.html = final_html
-
+        self._template_found = False
+        self._render_subject()
+        self._render_html()
+        self._render_txt()
+        if not self._template_found:
+            log.warning(f'Template {template} not used; possibly invalid name')
         return self
 
+    # this attempts to find all the possible templates to look for, considering language etc
+    # this assumes template_name is a base of filenames like: NAME_html.jinja2 and NAME_txt.jinja2
+    # and that NAME.jinja2 will be assumed to be html
+    # valid flavor = html, txt, subject
+    def _templates_to_try(self, flavor):
+        from app.modules.site_settings.models import SiteSetting
+
+        temps = []
+        if self.template_name is None:
+            return temps
+        langs = []
+        if self.language:
+            langs.append(self.language)
+        site_lang = SiteSetting.get_string('preferred_language', 'en_us')
+        if not self.language == site_lang:
+            langs.append(site_lang)
+        for lang in langs:
+            temps.append(f'email/{lang}/{self.template_name}_{flavor}.jinja2')
+            if flavor == 'html':  # also try flavorless if html
+                temps.append(f'email/{lang}/{self.template_name}.jinja2')
+        return temps
+
+    # this tries to find the best-fitting template
+    def _try_templates(self, flavor):
+        for temp in self._templates_to_try(flavor):
+            try:
+                rt = render_template(temp, **self.template_kwargs)
+                log.debug(f'Template flavor={flavor} matched {temp}')
+                self._template_found = True
+                return rt
+            except TemplateNotFound:
+                pass
+        return None
+
+    def _render_subject(self):
+        if self.subject:
+            return
+        self.subject = self._try_templates('subject')
+        if not self.subject:
+            self.subject = 'A message from Codex'
```

Comment:
should be in it's own jinja file for different languages and should contain something about which wildme it's coming from e.g. ACW rather than 'Codex'

Reply:
yeah, i do like this idea; if we are going to have a default for anything (and subject seems way more likely than body), we may as well give it a little more tweakability.   added `_find_default_subject()` and corresponding template.

this makes me wonder if in "the future" (post-mvp) these jinja templates will need to be either editable via the site-admin ui, or maybe overridden with SiteSetting values.  but anyway, thats for another day.